sycl/include/CL/sycl/detail/image_impl.hpp

```diff
@@ -8,150 +8,322 @@
+    range<2> temp = range<2>(RowPitch, SlicePitch);
+    return temp;
+  }
+
+  // Returns the size of the image storage in bytes
+  size_t get_size() const { return MSizeInBytes; }
 
-  size_t get_size() const { assert(!"Not implemented"); return 0;}
+  // Returns the total number of elements in the image
+  size_t get_count() const { return Range.size(); }
 
-  size_t get_count() const { assert(!"Not implemented"); return 0; }
+  // Returns the allocator provided to the image
+  AllocatorT get_allocator() const { return MAllocator; }
 
-  AllocatorT get_allocator() const { assert(!"Not implemented"); }
+  template <typename propertyT> bool has_property() const {
+    return MProps.has_property<propertyT>();
+  }
 
-  template <typename dataT, access::mode accessMode>
-  accessor<dataT, dimensions, accessMode, access::target::image>
-  get_access(handler &commandGroupHandler) {
-    assert(!"Not implemented");
+  template <typename propertyT> propertyT get_property() const {
+    return MProps.get_property<propertyT>();
   }
 
-  template <typename dataT, access::mode accessMode>
-  accessor<dataT, dimensions, accessMode, access::target::host_image>
-  get_access() {
-    assert(!"Not implemented");
+  // TODO: Implement this function.
+  void *allocateHostMem() override {
+    if (true)
+      throw cl::sycl::feature_not_supported(
+          "HostMemoryAllocation Function Not Implemented for image class");
+    return nullptr;
+    // Implementation of the pure virtual function.
   }
 
-  // template <typename Destination = std::nullptr_t>
-  // void set_final_data(Destination finalData = std::nullptr);
+  // TODO: Implement this function.
+  void *allocateMem(ContextImplPtr Context, bool InitFromUserData,
+                    cnri_event &OutEventToWait) override {
+    if (true)
+      throw cl::sycl::feature_not_supported(
+          "MemoryAllocation Function Not Implemented for image class");
+    return nullptr;
+    // Implementation of the pure virtual function.
+  }
+
+  MemObjType getType() const override { return MemObjType::IMAGE; }
+
+  // TODO: Implement this function.
+  void releaseHostMem(void *Ptr) override {
+    if (true)
+      throw cl::sycl::feature_not_supported(
+          "HostMemoryRelease Function Not Implemented for Image class");
+    return;
+    // Implementation of the pure virtual function.
+  }
+
+  // TODO: Implement this function.
+  void releaseMem(ContextImplPtr Context, void *MemAllocation) override {
+    if (true)
+      throw cl::sycl::feature_not_supported(
+          "MemoryRelease Function Not Implemented for Image class");
+    return;
+    // Implementation of the pure virtual function.
+  }
 
-  void set_write_back(bool flag) { assert(!"Not implemented"); }
-};
+private:
+  bool MHostPtrReadOnly = false;
+  AllocatorT MAllocator;
+  std::function<void(void)> MUploadDataFn = nullptr;
+  void *MUserPtr = nullptr;
+
+  property_list MProps;
```

Comment:
M here stands for member of class. So could you please either add the prefix to all class member of remove from all.